R/import-standalone-config.R

```diff
@@ -0,0 +1,241 @@
+#' CLI](https://docs.snowflake.com/en/developer-guide/snowflake-cli/connecting/configure-connections),
+#' or specifies them for a connection manually.
+#'
+#' # Common parameters
+#'
+#' The following is a list of common connection parameters. A more complete list
+#' can be found in [Snowflake's documentation for the Python
+#' Connector](https://docs.snowflake.com/en/developer-guide/python-connector/python-connector-api#label-snowflake-connector-methods-connect):
+#'
+#' - `account`: A Snowflake account identifier.
+#' - `user`: A Snowflake username.
+#' - `role`: The role to use for the connection.
+#' - `schema`: The default schema to use for the connection.
+#' - `database`: The default database to use for the connection.
+#' - `warehouse`: The default warehouse to use for the connection.
+#' - `authenticator`: The authentication method to use for the connection.
+#' - `private_key` or `private_key_file`: A path to a PEM-encoded private key
+#'    for key-pair authentication.
+#' - `private_key_file_pwd`: The passphrase for the private key, if any.
+#' - `token`: The OAuth token to use for authentication.
+#' - `token_file_path`: A path to an OAuth token to use for authentication.
+#' - `password`: The user's Snowflake password.
+#'
+#' @param name A named connection. Defaults to
+#'   `$SNOWFLAKE_DEFAULT_CONNECTION_NAME` if set, the `default_connection_name`
+#'   from the `config.toml` file (if present), and finally the `[default]`
+#'   section of the `connections.toml` file, if any. See [Snowflake's
+#'   documentation](https://docs.snowflake.com/en/developer-guide/python-connector/python-connector-connect#setting-a-default-connection)
+#'   for details.
+#' @param ... Additional connection parameters. See **Common parameters**.
+#' @param .config_dir The directory to search for a `connections.toml` and
+#'   `config.toml` file. Defaults to `$SNOWFLAKE_HOME` or `~/.snowflake` if that
+#'   directory exists, otherwise it falls back to a platform-specific default.
+#'   See [Snowflake's
+#'   documentation](https://docs.snowflake.com/en/developer-guide/python-connector/python-connector-connect#connecting-using-the-connections-toml-file)
+#'   for details.
+#'
+#' @returns An object of class `"snowflake_connection"`.
+#' @examples
+#' \dontrun{
+#' # Read the default connection parameters from an existing
+#' # connections.toml file:
+#' conn <- snowflake_connection()
+#'
+#' # Read a named connection from an existing connections.toml file:
+#' conn <- snowflake_connection(name = "default")
+#'
+#' # Override specific parameters for a connection:
+#' conn <- snowflake_connection(
+#'   schema = "myschema",
+#'   warehouse = "mywarehouse"
+#' )
+#' }
+#' @examples
+#' # Pass connection parameters manually, which is useful if there is no
+#' # connections.toml file. For example, to use key-pair authentication:
+#' conn <- snowflake_connection(
+#'   account = "myaccount",
+#'   user = "me",
+#'   private_key = "rsa_key.p8"
+#' )
+#' @export
```

Comment:
We do not want these names to become exported by rsconnect.

We have similar files from rlang, but those don't end up as exported.
https://github.com/rstudio/rsconnect/blob/main/R/import-standalone-obj-type.R